client/me/purchases/concierge-banner/index.jsx

```diff
@@ -42,86 +42,91 @@ class ConciergeBanner extends Component {
+								/>
+							),
+						},
 					}
 				);
 				buttonText = translate( 'Session dashboard' );
-				buttonHref = '/me/quickstart';
-				illustrationUrl = conciergeImage;
+				event = 'view-concierge-dashboard';
 				break;
-
+			case CONCIERGE_HAS_AVAILABLE_PURCHASED_SESSION:
 			case CONCIERGE_HAS_AVAILABLE_INCLUDED_SESSION:
-				headerText = translate( 'Looking for Expert Help?' );
-				mainText = translate(
-					'Get %(durationInMinutes)d minutes dedicated to the success of your site. Schedule your free 1-1 Quick Start Session with a Happiness Engineer!',
+				title = translate( 'You still have a Quick Start session available' );
+				description = translate(
+					`Schedule your {{supportLink}}Quick Start support session{{/supportLink}} and get one-on-one guidance from our expert Happiness Engineers to kickstart your site.`,
 					{
-						comment:
-							'Quick Start Session is a one-on-one video session between the user and our support staff.',
-						args: { durationInMinutes: 30 },
+						components: {
+							supportLink: (
+								<a
+									target={ '_blank' }
+									rel={ 'noreferrer' }
+									href="https://wordpress.com/discover-wordpress/2019/03/21/getting-the-most-out-of-our-business-concierge-service/"
+								/>
+							),
+						},
 					}
 				);
-				buttonText = translate( 'Schedule now' );
-				buttonHref = '/me/quickstart';
-				illustrationUrl = conciergeImage;
-				break;
-
-			case CONCIERGE_HAS_AVAILABLE_PURCHASED_SESSION:
-				headerText = translate( 'Our experts are waiting to help you' );
-				mainText = translate( 'Schedule your 1-1 Quick Start Session with a Happiness Engineer!', {
-					comment:
-						'Quick Start Session is a one-on-one video session between the user and our support staff.',
-				} );
-				buttonText = translate( 'Schedule now' );
-				buttonHref = '/me/quickstart';
-				illustrationUrl = conciergeImage;
+				buttonText = translate( 'Schedule a date' );
+				event = 'schedule-concierge-session';
 				break;
 		}
 
-		return { headerText, mainText, buttonText, buttonHref, illustrationUrl };
+		return { title, description, buttonText, event };
 	}
 
 	render() {
-		const { showPlaceholder } = this.props;
+		const { bannerType, showPlaceholder, nextAppointmentSiteId, siteId, availableSessions } =
+			this.props;
+
+		// if no appointment and not on a site then use the first site with a session available
+		// e.g. when viewing at /me/purchases
+		const quickStartSite = nextAppointmentSiteId || siteId || availableSessions[ 0 ];
 
 		if ( showPlaceholder ) {
 			return this.placeholder();
 		}
 
-		const { headerText, mainText, buttonText, buttonHref, illustrationUrl } =
-			this.getBannerContent();
+		const { buttonText, description, title, event } = this.getBannerContent();
+
+		const referrer = location.pathname.startsWith( '/me/purchases' )
+			? '/me/purchases'
+			: '/purchases/subscriptions/:site';
```

Comment:
This doesn't feel right to me. I think we should not mediate the current path directly, and we should probably not be using `location` directly either.

I _think_ it will probably be sufficient for us to handle this via the `getCurrentRoute()` and `getCurrentRouteParameterized()` selectors (where we'd use the second when we `sited` is defined).

Reply:
I've changed this as suggested.